plexpy/datafactory.py

```diff
@@ -779,3 +779,190 @@ def delete_all_user_history(self, user_id=None):
+                'media_index, parent_media_index ' \
+                'FROM session_history_metadata ' \
+                'WHERE {0} = ? ' \
+                'GROUP BY {1} '
+
+        # get grandparent_rating_keys
+        grandparents = {}
+        result = monitor_db.select(query=query.format('grandparent_rating_key', 'grandparent_rating_key'),
+                                   args=[grandparent_rating_key])
+        for item in result:
+            # get parent_rating_keys
+            parents = {}
+            result = monitor_db.select(query=query.format('grandparent_rating_key', 'parent_rating_key'),
+                                       args=[item['grandparent_rating_key']])
+            for item in result:
+                # get rating_keys
+                children = {}
+                result = monitor_db.select(query=query.format('parent_rating_key', 'rating_key'),
+                                           args=[item['parent_rating_key']])
+                for item in result:
+                    key = item['media_index']
+                    children.update({key: {'rating_key': item['rating_key']}})
+
+                key = item['parent_media_index'] if match_type == 'index' else item['parent_title']
+                parents.update({key:
+                                {'rating_key': item['parent_rating_key'],
+                                 'children': children}
+                                })
+
+            key = 0 if match_type == 'index' else item['grandparent_title']
+            grandparents.update({key:
+                                 {'rating_key': item['grandparent_rating_key'],
+                                  'children': parents}
+                                 })
+
+        key_list = grandparents
+        
+        return key_list
+
+    def update_rating_key(self, old_key_list='', new_key_list='', media_type=''):
+        monitor_db = database.MonitorDatabase()
+
+        # function to map rating keys pairs
+        def get_pairs(old, new):
+            pairs = {}
+            for k, v in old.iteritems():
+                if k in new:
+                    if v['rating_key'] != new[k]['rating_key']:
+                        pairs.update({v['rating_key']: new[k]['rating_key']})
+                    if 'children' in old[k]:
+                        pairs.update(get_pairs(old[k]['children'], new[k]['children']))
+
+            return pairs
+
+        # map rating keys pairs
+        mapping = {}
+        if old_key_list and new_key_list:
+            mapping = get_pairs(old_key_list, new_key_list)
+        
+        if mapping:
+            logger.info(u"PlexPy DataFactory :: Updating rating keys in the database.")
+            for old_key, new_key in mapping.iteritems():
+                # check rating_key (3 tables)
+                monitor_db.action('UPDATE session_history SET rating_key = ? WHERE rating_key = ?', [new_key, old_key])
```

Comment:
Just a formatting thing here, could you just make sure that the lines don't get too long here. PEP8 default is 80 characters per line but I find that too short - I use 120 which is the default on PyCharm.